### PR TITLE
1714: Updating texter filter in Campaign/texters screen

### LINKED ISSUE #1714: Bug: New "Texting Alias" value interferes with search for texters to assign
**Describe the bug**
Spoke assigners typically receive a request from a texter in Slack (e.g. "John Smith") and then search for that same name in the Spoke text assigning panel. But if a "texting alias" has been set by the texter in Spoke, the a First+Last search will not find the texter. This make it appear, incorrectly, that the texter does not have a Spoke account and cannot be assigned texts.

This is especially problematic when the texter has a common first name (like "John"), and the assigner is habituated to add the last name when searching in order to narrow down the search results.

The text assigners search string should be matched against the Spoke First + Last name without the parenthesized alias.

**To Reproduce**
1. Create an account in Spoke with the fields "First name", "Last name", and "Texting Alias (optional)" all filled in, e.g. "John", "Smohn", and "Pat", respectively.
2. Select any campaign, click "Edit", then select the "Texters" panel.
3. In the "Search for texters to assign" field, type "John Smohn"
4. The texter will not be located.
5. Now in the "Search for texters to assign" field, type "John (P"
6. The texter will be located.

**Expected behavior**
The text entered in the "Search for texters to assign" field should be matched only against the "First name" and "Last name" fields of Spoke accounts. The Texting Alias field value should be used only for display (in the search results and in the list of texters who have been assigned) and for inserting into text messages. It should not be searched on in the "Assign Text" panel.

(It should, however, be searchable in the "People" page of Spoke, along with First name, Last name, and Email. Currently it isn't.)

**Screenshots**
I created a Spoke account with the following profile values:
  First name: John
  Last name: Smith
  Texting alias: Pat
  I then tried searching on "John Smith" and "John (P". The screen shots below show the results of those two searches. Only the latter gives a result, although it is the former that should give the result.

Screen Shot 2020-08-02 at 9 24 34 PM
Screen Shot 2020-08-02 at 9 24 50 PM

**Platform (please complete the following information):**
 - OS: MacOS 10.13.6
 - Browser: Firefox
 - Desktop or Mobile? Desktop
This bug appears to occur on all platforms and in all browsers.

**Additional context**
Is this really a bug or is this a feature change request? All of the text assigners on the MoveOn Text Team find the current functionality confusing and a hindrance, so I'm considering it a bug.

### PR DESCRIPTION
**Pull request by bdatkins**
# Fixes #1714

## Description

_Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change, and any blockers that make your change a WIP_

# Checklist:

- [X] I have manually tested my changes on desktop and mobile
- [X] The test suite passes locally with my changes
- [X] If my change is a UI change, I have attached a screenshot to the description section of this pull request
- [X] [My change is 300 lines of code or less](https://github.com/MoveOnOrg/Spoke/blob/main/CONTRIBUTING.md#your-first-code-contribution), or has a documented reason in the description why it’s longer
- [X] I have made any necessary changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] My PR is labeled [WIP] if it is in progress

Screen Shot 2020-08-16 at 3 27 48 PM

### CODE REVIEW

**Comment by bdatkins** [comment]
Looking for suggestions on what tests to add since it's a UI feature (@ibrand?)

File: src/components/CampaignTextersForm.jsx [hunk 1]
```diff
@@ -279,13 +279,12 @@ export default class CampaignTextersForm extends React.Component {
...
     const filter = (searchText, key) =>
-      key === "allTexters"
-        ? true
-        : AutoComplete.caseInsensitiveFilter(searchText, key);
+      key === "allTexters" ? true : AutoComplete.fuzzyFilter(searchText, key);
 
     const autocomplete = (
       <AutoComplete
         ref="autocomplete"
+        s
```
Comment (schuyler1d) [inline src/components/CampaignTextersForm.jsx]: is this a typo or a real property?

**Review by schuyler1d — DISMISSED** [review]
I think we can skip tests for this one as the main function is in AutoComplete

File: src/components/CampaignTextersForm.jsx (hunk 1 above)
Comment (bdatkins) [reply src/components/CampaignTextersForm.jsx]: Good catch @schuyler1d  -- fixed the typo

**Review by schuyler1d — APPROVED** [review]
:+1:

**Pull request merged**